recipes/libcurl/all/conanfile.py

```diff
@@ -170,7 +170,7 @@ def build_requirements(self):
            tools.os_info.detect_windows_subsystem() != "msys2":
             self.build_requires("msys2/20200517")
```

Comment:
i would say that is too lax mingw detection. This leads to msys2 being installed as build requirements  even though the library is compiled for android for example. #4277

Reply:
probably, needs also `settings.os == "Windows"`? and probably, `is_mingw` property needs to check `settings.compiler == "gcc"` as well?

Reply:
also don't forget this comment: https://github.com/conan-io/conan-center-index/pull/4457#discussion_r572891032

Reply:
@SSE4 I guess so.
https://github.com/conan-io/conan-center-index/blob/4cfe106fab3bf1593d39f802a7a5a69ee68250b3/recipes/libcurl/all/conanfile.py#L77

 return self.settings.os == "Windows" and self.settings.compiler == "gcc"

Reply:
@SpaceIm _is_using_cmake_build is always true when cross compile for Android on Windows

        if not self._is_using_cmake_build:
        ....
        elif self._is_win_x_android:

https://github.com/conan-io/conan-center-index/blob/4cfe106fab3bf1593d39f802a7a5a69ee68250b3/recipes/libcurl/all/conanfile.py#L84-L85

Reply:
there is a couple of another weirdness in the recipe:
- CMake build is used only for some platforms (e.g. Android)
- Android build enforces ninja (shouldn't it work with any generator?)

anyway, I am not going to address this right now. let's just merge what we have already, otherwise, it will never end :) future PR may address these problems.

Reply:
No problem, building for Android on Windows is just my hobby project. :-) However what I have never understood why the autoconfig build is still used when the CMake build is available.